src/EnergyPlus/OutputProcessor.hh

```diff
@@ -74,6 +74,51 @@ struct EnergyPlusData;
 
 namespace OutputProcessor {
 
+    enum class eTimeStepType : int
+    {
+        Zone = 0,
+        HeatBalance = 1,
+        HVAC = 2,
+        System = 3,
+        Plant = 4
+    };
+    std::array<std::string_view, 5> constexpr sTimeStepType = {"Zone", "HeatBalance", "HVAC", "System", "Plant"};
+    inline eTimeStepType getTimeStepTypeEnum(std::string const &s)
+    {
+        if (s == "Zone") {
```

Comment:
Can do this by looping over `sTimeStepType` array. Actually, this should not be a eTimeStepType function. There should be a single utility function `int getEnumerationValue(std::string_view s, gsl::span, sList)` and then this would be replaced by a call `static_cast<eTimeStepType>getEnumerationValue(s, sTimeStepType)`.